frontend/ScanUI/components/User/SignUp.js

```diff
@@ -1,11 +1,131 @@
-import React from "react ";
-import { View, TouchableOpacity, Text } from "react-native";
+import { React, useState } from "react";
+import {
+  View,
+  TouchableOpacity,
+  Text,
+  TextInput,
+  SafeAreaView,
+} from "react-native";
+import Styles from "../Styles";
 
-const signUp = () => {
+const signUp = ({ navigation }) => {
+  const [name, setName] = useState("");
+  const [surname, setSurname] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [email, setEmail] = useState("");
+  const isEmpty = () => {
+    return (
+      name === "" ||
+      surname === "" ||
+      username === "" ||
+      password === "" ||
+      confirmPassword === ""
+    );
+  };
+  const validateLength = () => {
+    if (password.length > 3) {
+      return false;
+    }
+    return true;
+  };
+  const isValidEmail = () => {
+    if (email.includes("@")) {
+      return true;
+    }
+    return false;
+  };
```

Comment:
Create a separate folder for validations and create a file for this and call it whenever you need it